src/index.ts

```diff
@@ -384,6 +384,114 @@ export class MidiIn {
   }
 }
 
+/**
+ * Function to call when a MIDI note-on event is received from a multichannel controller (e.g. Lumatone).
+ * Attack velocity is from 0 to 127. Channel is from 0 to 15.
+ * Must return a note-off callback (e.g. for turning off your synth).
+ */
+export type MultiChannelNoteOnCallback = (index: number, rawAttack: number, channel: number) => NoteOff;
+
+/**
+ * Wrapper for webmidi.js input.
+ * Listens on multiple channels, maps incoming channels to equaves around a central channel
+ */
+export class MultiChannelMidiIn extends MidiIn {
+  callback: MultiChannelNoteOnCallback;
+  channels: Set<number>;
+  /** Note-off map from (noteNumber + midiChannel * 128) to callbacks.  */
+  private noteOffMap: Map<number, (rawRelease: number) => void>;
+  private _noteOn: (event: NoteMessageEvent) => void;
+  private _noteOff: (event: NoteMessageEvent) => void;
+  log: (msg: string) => void;
+
+  /**
+   * Construct a new wrapper for a webmidi.js input device.
+   * @param callback Function to call when a note-on event is received on any of the available channels.
+   * @param channels Channels to listen on.
+   * @param log Logging function.
+   */
+  constructor(
+    callback: MultiChannelNoteOnCallback,
+    channels: Set<number>,
+    log?: (msg: string) => void
+  ) {
+    this.callback = callback;
+    this.channels = channels;
+    this.noteOffMap = new Map();
+
+    this._noteOn = this.noteOn.bind(this);
+    this._noteOff = this.noteOff.bind(this);
+
+    if (log === undefined) {
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars
+      this.log = msg => {};
+    } else {
+      this.log = log;
+    }
+  }
+
+  /**
+   * Make this wrapper (and your callback) respond to note-on/off events from this MIDI input.
+   * @param input MIDI input to listen to.
+   */
+  listen(input: Input) {
+    input.addListener('noteon', this._noteOn);
+    input.addListener('noteoff', this._noteOff);
+  }
+
+  /**
+   * Make this wrapper (and your callback) stop responding to note-on/off events from this MIDI input.
+   * @param input MIDI input that was listened to.
+   */
+  unlisten(input: Input) {
+    input.removeListener('noteon', this._noteOn);
+    input.removeListener('noteoff', this._noteOff);
+  }
+
+  private noteOn(event: NoteMessageEvent) {
+    if (!this.channels.has(event.message.channel)) {
+      return;
+    }
+    const noteNumber = event.note.number;
+    const attack = event.note.attack;
+    const rawAttack = event.note.rawAttack;
+    this.log(
+      `Midi note on ${noteNumber} at velocity ${attack} on channel ${event.message.channel}`
+    );
+    const noteOff = this.callback(noteNumber, rawAttack);
```

Comment:
What is this code doing in this PR and why is the channel not passed to the callback? If you're doing the channel-to-pitch conversion over at Scale Workshop, we could just extend the the base class to include channel information. Just need to check that it's OK by TypeScript to have callbacks that ignore the channel.

Reply:
sorry, my mistake, I was experimenting.... this code was not meant to be part of my PR -- all I was suggesting is the channel -1 thing - if you think it is irrelevant we can close this